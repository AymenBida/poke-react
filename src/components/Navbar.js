import { Link } from 'react-router-dom';

const Navbar = () => (
  <nav>
    <ul>
      <li><h1>PokeReact</h1></li>
      <li><Link to="/">Home</Link></li>
      <li><Link to="/pokemon">Pokemon</Link></li>
      <li><Link to="/about">About</Link></li>
    </ul>
  </nav>
);

export default Navbar;
