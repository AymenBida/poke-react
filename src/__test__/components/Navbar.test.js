import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../../components/Navbar';

describe('Navbar', () => {
  it('renders correctly', () => {
    const element = render(<BrowserRouter><Navbar /></BrowserRouter>);
    expect(element).toMatchSnapshot();
  });
});
