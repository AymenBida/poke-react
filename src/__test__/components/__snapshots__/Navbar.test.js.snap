// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar navbar-expand-lg navbar-light bg-light px-3"
      >
        <div
          class="container-fluid"
        >
          <h1
            class="navbar-brand mb-0 title"
          >
            PokeReact
          </h1>
        </div>
        <div
          class="navbar-nav"
        >
          <a
            class="nav-link"
            href="/"
          >
            Home
          </a>
          <a
            class="nav-link"
            href="/pokemon"
          >
            Pokemon
          </a>
          <a
            class="nav-link"
            href="/about"
          >
            About
          </a>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar navbar-expand-lg navbar-light bg-light px-3"
    >
      <div
        class="container-fluid"
      >
        <h1
          class="navbar-brand mb-0 title"
        >
          PokeReact
        </h1>
      </div>
      <div
        class="navbar-nav"
      >
        <a
          class="nav-link"
          href="/"
        >
          Home
        </a>
        <a
          class="nav-link"
          href="/pokemon"
        >
          Pokemon
        </a>
        <a
          class="nav-link"
          href="/about"
        >
          About
        </a>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
